### PR TITLE
Imagestream restore plugin

### PR DESCRIPTION
**Pull request by sseago**
Copies image from migration registry for local images.
Discards item on restore, since:
1) for local images, image import recreates imagestream
2) for remote images, imagestreamtag restore recreates imagestream

### CODE REVIEW

File: velero-plugins/imagestream/restore.go [hunk 1]
```diff
@@ -18,22 +24,87 @@ func (p *RestorePlugin) AppliesTo() (velero.ResourceSelector, error) {
...
+	migrationRegistry := input.Restore.Annotations[common.MigrationRegistry]
+	if len(migrationRegistry) == 0 {
+		return nil, errors.New("migration registry not found for annotation \"openshift.io/migration\"")
+	}
+	p.Log.Info(fmt.Sprintf("backup internal registry: %#v", backupInternalRegistry))
+	p.Log.Info(fmt.Sprintf("restore internal registry: %#v", internalRegistry))
 
-	return velero.NewRestoreItemActionExecuteOutput(input.Item), nil
+	for _, tag := range imageStreamUnmodified.Status.Tags {
+		p.Log.Info(fmt.Sprintf("tag: %#v", tag.Tag))
```
Comment (dymurray) [inline velero-plugins/imagestream/restore.go]: Remove this or maybe a more descriptive log statement:
```
p.Log.Info(fmt.Sprintf("Restoring tag: %v"))
```

File: velero-plugins/imagestream/restore.go [hunk 2]
```diff
@@ -18,22 +24,87 @@ func (p *RestorePlugin) AppliesTo() (velero.ResourceSelector, error) {
...
+		copyToTag := true
+		if specTag != nil && specTag.From != nil {
+			p.Log.Info(fmt.Sprintf("image tagged: %s, %s", specTag.From.Kind, specTag.From.Name))
+			// we have a tag.
+			copyToTag = false
+		}
+		// Iterate over items in reverse order so most recently tagged is copied last
+		for i := len(tag.Items) - 1; i >= 0; i-- {
+			dockerImageReference := tag.Items[i].DockerImageReference
+			localImage := strings.HasPrefix(dockerImageReference, backupInternalRegistry)
```
Comment (dymurray) [inline velero-plugins/imagestream/restore.go]: Could optionally do:
```
if strings.HasPrefix(dockerImageReference, backupInternalRegistry) {
```

**Review by dymurray — APPROVED** [review]

File: velero-plugins/imagestream/restore.go (hunk 2 above)
Comment (sseago) [reply velero-plugins/imagestream/restore.go]: I think in an earlier iteration of my backup plugin I used the variable more than once. This assignment isn't needed anymore.

**Comment by sseago** [comment]
Updated to incorporate feedback.

**Pull request merged**